totalRP3/Core/Vanilla/CompanionUtilOverrides.lua

```diff
@@ -0,0 +1,322 @@
+	-- can't localize the summoned name of creatures ahead of time.
+	--
+	-- Instead, query the creature ID of the unit and map that back to a name.
+
+	local unitGUID = UnitGUID(unitToken);
+	local creatureID = tonumber((select(6, string.split("-", unitGUID or "", 7))));
+	local staticPetInfo = staticCompanionPetsByCreatureID[creatureID];
+
+	if staticPetInfo then
+		return staticPetInfo.name;
+	end
+end
+
+
+function TRP3_CompanionUtil.GetSummonedCompanionPetID()
+	if TRP3_Companions then
+		return TRP3_Companions.summonedPetID;
+	end
+end
+
+function TRP3_CompanionUtil.GetMountIDs()
+	return GetKeysArray(staticMountsByID);
+end
+
+function TRP3_CompanionUtil.GetMountInfo(mountID)
+	local staticMountInfo = staticMountsByID[mountID];
+
+	if not staticMountInfo then
+		return;
+	end
+
+	return {
+		id = staticMountInfo.mountID,
+		name = staticMountInfo.name,
+		icon = staticMountInfo.icon,
+		description = "",
+		isCollected = nil,  -- Indeterminate; would require a bag and bank scan.
+		spellID = staticMountInfo.spellID,
+	};
+end
+
+function TRP3_CompanionUtil.GetMountSpellID(mountID)
+	local staticMountInfo = staticMountsByID[mountID];
+
+	if staticMountInfo then
+		return staticMountInfo.spellID;
+	end
+end
+
+local function GetUnitAuraSpells(unitToken, filter)
+	-- Inner 'select' here is to skip the leading 'continuationToken' return.
+	local auraSlots = { select(2, C_UnitAuras.GetAuraSlots(unitToken, filter)) };
+	local appliedAuras = {};
+
+	for slotIndex = 1, #auraSlots do
+		local auraInfo = C_UnitAuras.GetAuraDataBySlot(unitToken, auraSlots[slotIndex]);
+
+		if auraInfo then
+			appliedAuras[auraInfo.spellId] = true;
+		end
+	end
+
+	return appliedAuras;
+end
+
+function TRP3_CompanionUtil.GetSummonedMountID()
+	if not IsMounted() then
+		return nil;
+	end
+
+	-- Is the last mount that we tested still the current summon?
+	if cachedSummonedMountID ~= nil then
+		local spellID = TRP3_CompanionUtil.GetMountSpellID(cachedSummonedMountID);
+
+		if not C_UnitAuras.GetPlayerAuraBySpellID(spellID) then
+			cachedSummonedMountID = nil;
+		end
+	end
+
+	-- Has our cache been invalidated? If so, detect from applied auras.
+	if cachedSummonedMountID == nil then
+		local appliedAuras = GetUnitAuraSpells("player", "HELPFUL|PLAYER|CANCELABLE");
+
+		for _, mountID in ipairs(TRP3_CompanionUtil.GetMountIDs()) do
+			local spellID = TRP3_CompanionUtil.GetMountSpellID(mountID);
+
+			if appliedAuras[spellID] then
+				cachedSummonedMountID = mountID;
+				break;
+			end
+		end
+	end
+
+	return cachedSummonedMountID;
+end
+
+-- On clients that don't have a pet or mount journal, we statically embed
+-- a database with just enough data to allow us to build our own fake version
+-- of the journals in-game.
+--
+-- This static data doesn't include localized strings, and needs to be
+-- filtered based on what's actually available in the client. To do this we
+-- need to do some sparse loads for item and spell data prior to registering
+-- the companion.
```

Comment:
One thing not documented here - previously we filtered out companions and mounts that didn't exis in a client via C_Item.DoesItemExistByID and C_Spell.DoesSpellExist.

The former of these functions has been "broken" for a while and returned false only if the spell ID were wholly invalid.

Now, we don't use those functions - instead, we trigger a spell/item load request and only register the companion in our static data tables if we get a successful result.